prdoc/pr_4378.prdoc

```diff
@@ -0,0 +1,15 @@
+# Schema: Polkadot SDK PRDoc Schema (prdoc) v1.0.0
+# See doc at https://raw.githubusercontent.com/paritytech/polkadot-sdk/master/prdoc/schema_user.json
+
+title: "Use `versioned_type!` macro for `VersionedXcm`"
+
+doc:
+  - audience: Runtime Dev
+    description: |
+      Currently, all other versioned types in the `staging-xcm` are created using `versioned_type!` macro, except for `VersionedXcm`. 
+      This PR changes `versioned_type` macro so that it can be used for `VersionedXcm` as well. This is done by adding optional
```

Comment:
```suggestion
      This PR changes `versioned_type` macro so that it can be used for `VersionedXcm` as well, reducing a lot of duplication. This is done by adding optional
```